website/docs/language/style.mdx

```diff
@@ -0,0 +1,729 @@
+
+Monolithic repositories make the repository the single source of truth, where the repository contains every change made to the infrastructure. This lowers the maintenance burden by reducing the number of repositories you must manage and track. If your organization requires a monolithic approach, Terraform Cloud and Terraform Enterprise let you scope a workspace to a specific directory in a repository, simplifying your workflows.
+
+## Branching strategy
+
+To collaborate on your Terraform code, we recommend using the [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow). This approach uses short-lived branches to help your team quickly review, test, and merge changes to your code. To make changes to your code, you would:
+
+1. Create a new branch from your main branch
+1. Write, commit, and push your changes to the new branch
+1. Create a pull request
+1. Review the changes with your team
+1. Merge the pull request
+1. Delete the branch
+
+Terraform Cloud and Terraform Enterprise can run [speculative plans for pull requests](/terraform/cloud-docs/run/ui#speculative-plans-on-pull-requests). These speculative plans run automatically when you create or update a pull request, and you can use them to see the effect that your changes will have on your infrastructure before you merge them to your main branch. When you merge your pull request, Terraform Cloud will start a new run to apply these changes.
+
+## Multiple environments
+
+We recommend that your repository's `main` branch be the source of truth for all environments. For Terraform Cloud and Terraform Enterprise users, we recommend that you use separate workspaces for each environment. For larger codebases, we recommend that you split your resources across multiple workspaces to prevent large state files and limit unintended consequences from changes. For example, you could structure your code as follows:
+
+<CodeBlockConfig hideClipboard>
+
+```
+.
+├── compute
+│   ├── main.tf
+│   ├── outputs.tf
+│   └── variables.tf
+├── database
+│   ├── main.tf
+│   ├── outputs.tf
+│   └── variables.tf
+└── networking
+    ├── main.tf
+    ├── outputs.tf
+    └── variables.tf
+```
+
+</CodeBlockConfig>
+
+In this scenario, you would create three workspaces per environment. For example, your production environment would have a "prod-compute", "prod-database", and "prod-networking" workspace. Read more about [Terraform workspace and project best practices](/well-architected-framework/operational-excellence/operational-excellence-workspaces-projects). 
+
+If you do not use Terraform Cloud or Terraform Enterprise, we recommend that you use modules to encapsulate your configuration, and use a directory for each environment. The configuration in each of these directories would call the local modules, each with parameters specific to their environment. This also lets you maintain separate variable and backend configurations for each environment. 
```

Comment:
i would draw the connection between directories and state files. also mention that at some point, env-level state files become very risky.